cpp/open3d/geometry/PointCloud.cpp

```diff
@@ -158,6 +158,44 @@ std::vector<double> PointCloud::ComputePointCloudDistance(
     return distances;
 }
 
+PointCloud &PointCloud::RemoveDuplicatedPoints() {
+    bool has_normals = HasNormals();
+    bool has_colors = HasColors();
+    bool has_covariances = HasCovariances();
+    size_t old_points_num = points_.size();
+    size_t k = 0;
+
+    typedef std::tuple<double, double, double> Coordinate3;
+    std::unordered_map<Coordinate3, size_t, utility::hash_tuple<Coordinate3>>
+            point_to_old_index;
+
+    for (size_t i = 0; i < old_points_num; i++) {
+        Coordinate3 coord =
+                std::make_tuple(points_[i](0), points_[i](1), points_[i](2));
+        if (point_to_old_index.find(coord) == point_to_old_index.end()) {
+            point_to_old_index[coord] = i;
+            points_[k] = points_[i];
+            if (has_normals) normals_[k] = normals_[i];
+            if (has_colors) colors_[k] = colors_[i];
+            // estimate covariance later (copying covariances won't work)
+            k++;
+        }
+    }
+
+    points_.resize(k);
+    if (has_normals) normals_.resize(k);
```

Comment:
Normals are computed using covariance, so it might change as well. We should compute the same again. 

Also, check `has_covariances` before `has_normals`. As it's almost free to compute normals when we have covariances.